src/main/java/com/suse/saltstack/netapi/client/SaltStackClient.java

```diff
@@ -97,14 +97,18 @@ public Token login(String username, String password)
      * @return authentication token as {@link Token}
      * @throws SaltStackException if anything goes wrong
      */
-    public Token login(String username, String password, String eauth)
+    public Token login(final String username, final String password, final String eauth)
             throws SaltStackException {
-        JsonObject json = new JsonObject();
-        json.addProperty("username", username);
-        json.addProperty("password", password);
-        json.addProperty("eauth", eauth);
+        Map<String, String> kwargs = new LinkedHashMap() {
```

Comment:
Thanks for comment. Maybe I'm missing something, but I believe finalization of arguments doesn't leak outside of the method. It only means that arguments aren't overwriten inside the method. You can freely pass nonfinal fields to a method with final args. Isn't that correct?